backend/apps/xbin/apis/downloadfile.js

```diff
@@ -56,9 +57,9 @@ function _sendError(servObject, unauthorized) {
 async function _zipDirectory(path) {
     return new Promise((resolve, reject) => {
         const tempFilePath = utils.getTempFile("zip"); const out = fs.createWriteStream(tempFilePath);
-        const arhiver = archiver("zip", { zlib: { level: 9 }});
-        archiver.on("end", _=>resolve(tempFilePath)); archiver.on("error", err=>reject(err));
-        arhiver.directory(path, false).pipe(out, {end:true}); archive.finalize();
+        const archive = archiver("zip", { zlib: { level: 9 }});
+        out.on("close", _=>resolve(tempFilePath)); archive.on("error", err=>reject(err));
```

Comment:
Shouldn't this be archive.on("close")???

Reply:
archive.on("end") fired before out.on("close") during testing resulting in incomplete file downloads, that's why resolved the promise on close of writestream.